app/com/gu/floodgate/contentsource/ContentSource.scala

```diff
@@ -1,13 +1,20 @@
 package com.gu.floodgate.contentsource
 
 import java.util.UUID
+import com.gu.floodgate.reindex.DateParameters
 import play.json.extra.JsonFormat
 
 /*
  * Used for creation as we generate the ID server side and do not expect the client to provide it.
  */
 @JsonFormat
-case class ContentSourceWithoutId(appName: String, description: String, reindexEndpoint: String, environment: String)
+case class ContentSourceWithoutId(
+  appName: String,
+  description: String,
+  reindexEndpoint: String,
+  environment: String,
+  authType: String,
+  apiKey: Option[String])
```

Comment:
Does it need to be a separate field? Couldn't the user just include the api key param in the endpoint URL? e.g. `http://my-service/reindex?key=foo`. This means they can name the URL param however they like.